### ISSUE TITLE
cputc() and lowercase letters on non-enhanced Apple2

### ISSUE THREAD

**Issue by varna9000** [cc65/cc65#2456]
I see that cputc() for apple2 target [checks if target is apple2 enhanced](https://github.com/cc65/cc65/blob/0541b65aa48ea2508b218ff7097911e741b4bc12/libsrc/apple2/cputc.s#L75) in order to display lowercase letters.

I have an Apple2e clone (Pravetz 8C), which is not enhanced, but supports lowercase and 80columns (much as the not enhanced apple2e does). I can't seem to find a way to use any string printing function with target apple2 so I can print lower letters. If I use target apple2enh, it just doesn't work because it's just not an a2 enhanced. `islower()` and `isupper()` return correctly the case when I read the keyboard with `cgetc()`, but printing lowercase doesn't work.

**Comment by polluks** [comment]
Back in the day, when we defined apple2 and apple2enh we want a lowest common divisor platform and pushing the 65C02. Sorry, you are on your own.

**Comment by varna9000** [comment]
That's fine. I was just wondering if I'm missing something. I'll remove the target check from the function and re-compile it just for my purpose. Thanks for the clarification.

**Comment by mrdudz** [comment]
So this cant be fixed?

**Comment by polluks** [comment]
I was thinking of c64-soft80.o, maybe we can replace some conio functions by another object.

**Comment by colinleroy** [comment]
I think that this issue is fixed since the introduction of allow_lowercase() (#2474) (for lowercase), #2662 (auto lowercase), and #2641 (for 80cols).
The apple2 target now supports 80 columns and lowercase characters with automatic detection of capabilities in most cases; for the Pravetz 8C, a manual call to allow_lowercase(1) will be required.

**Comment by mrdudz** [comment]
I'm not sure how to even verify this stuff (does this work in emulation?)

@oliverschmidt please check

**Comment by oliverschmidt** [comment]
I'm a bit unsure what to write. @colinleroy's statement is correct in every aspect. And that comes as no surprise at all because he has implemented the changes in question.

And yes, this works in emulation. AppleWin can both emulate an Apple //e as well as an Enhanced Apple //e:
<img width="347" height="219" alt="Image" src="https://github.com/user-attachments/assets/9f223306-5d15-4f21-9114-492f562bfada" />

Those modes were used to verify the changes in question. So if the Pravetz 8C correctly identifies itself as Apple //e, then lowercase output is supposed to work as desired by the OP by simply building for the apple2 target.

The identification is based on https://mirrors.apple2.org.za/apple.cabi.net/FAQs.and.INFO/A2.TECH.NOTES.ETC/A2.CLASSIC.TNTS/a2misc007(1).htm and implemented in https://github.com/cc65/cc65/blob/master/libsrc/apple2/get_ostype.s

To summarize: I believe that this issue can be considered fixed with the changes in question.

**Comment by colinleroy** [comment]
> So if the Pravetz 8C correctly identifies itself as Apple //e

I think it identifies itself as a ][+, in which case the reporter (@varna9000) will have to call allow_lowercase(1). But I think we can consider that issue fixed as we won't be able to do more about it...

**Comment by mrdudz** [comment]
ok, fine then :)

**Comment by varna9000** [comment]
I'll test this on real hardware Pravetz 8C and report back. Thanks for fixing this!

**Comment by varna9000** [comment]
Hi, apologies for taking so long to test this. I confirm that with `apple2` target, I can now write small letters on `Pravetz 8C` (not enhanced apple//e). I've tested also the 80col mode at the same time.

![Image](https://github.com/user-attachments/assets/da094054-ae4a-4eb7-874c-066f6ec027ca)
